d3ploy/no_inst.py

```diff
@@ -120,39 +120,53 @@ def __init__(self, *args, **kwargs):
     def enter_notify(self):
         super().enter_notify()
         lib.TIME_SERIES_LISTENERS[self.supply_commod].append(self.extract_supply)
-        lib.TIME_SERIES_LISTENERS[self.demand_commod].append(self.extract_demand)   
-  
+        lib.TIME_SERIES_LISTENERS[self.demand_commod].append(self.extract_demand)
+
     def tick(self):
         """
         This is the tock method for the institution. Here the institution determines the difference
-        in supply and demand and makes the the decision to deploy facilities or not.     
+        in supply and demand and makes the the decision to deploy facilities or not.
         """
         time = self.context.time
         diff, supply, demand = self.calc_diff(time-1)
         if  diff < 0:
             proto = random.choice(self.prototypes)
             ## This is still not correct. If no facilities are present at the start of the
-            ## simulation prod_rate will still return zero. More complex fix is required.            
+            ## simulation prod_rate will still return zero. More complex fix is required.
             if proto in self.fac_supply:
                 prod_rate = self.fac_supply[proto]
             else:
-                print("No facility production rate available for " + proto)                
+                print("No facility production rate available for " + proto)
             number = np.ceil(-1*diff/prod_rate)
             for i in range(int(number)):
                 self.context.schedule_build(self, proto)
                 i += 1
+        if diff > 0:
```

Comment:
I think greater than 0 is a bad choice here. It should allow for some error in the over production. 

Maybe something like if diff > 0 + (2 x min production rate) or something?

Reply:
I agree some smarter decision should be made here.

Also, I was thinking that it should only decommission facilities when the oversupply continues for a certain timestep (an internal `count` variable that counts the length of continuous oversupply).

Reply:
Yeah, this is why decommissioning is very difficult. I think the behavior might vary from facility to facility. For instance a reactor might not decommission because it can't get fuel in a time step, but a mine might.

Reply:
I guess the best way is for the user to have the option of setting the `count`.